### PR TITLE
Updates for 0.12

### PR DESCRIPTION
**Pull request by paulyoung**
This is basically done except that `Locale` was removed in [`purescript-datetime` `v4.0.0`](https://github.com/purescript/purescript-datetime/releases/tag/v4.0.0).

> Removed `Locale` - it was a glorified `Tuple` without any useful extra functionality

I'm putting this up to discuss the way forward.

### CODE REVIEW

**Comment by paulyoung** [comment]
Since the whole idea with `purescript-precise-datetime` has been to mimic `purescript-datetime`, I think it makes sense to also remove it.

**Comment by natefaubion** [comment]
I think we should remove it as well.

**Comment by paulyoung** [comment]
I think anything depending on the things relating to `Locale` could reimplement what's being removed here to keep things working for the time being.

File: bower.json [hunk 1]
```diff
@@ -12,31 +12,29 @@
...
-    "purescript-numbers": "^5.0.0",
-    "purescript-decimals": "^3.4.0"
+    "purescript-arrays": "^5.0.0",
+    "purescript-bigints": "^4.0.0",
+    "purescript-console": "^4.1.0",
+    "purescript-datetime": "^4.0.0",
+    "purescript-either": "^4.0.0",
+    "purescript-enums": "^4.0.0",
+    "purescript-foldable-traversable": "^4.0.0",
+    "purescript-formatters": "slamdata/purescript-formatters#ee2a7e93dd124a71e188a75b002289bfffe4618e",
```
Comment (paulyoung) [inline bower.json]: We should wait on this.

File: bower.json [hunk 2]
```diff
@@ -12,31 +12,29 @@
...
+    "purescript-js-date": "^6.0.0",
+    "purescript-lists": "^5.0.0",
+    "purescript-maybe": "^4.0.0",
+    "purescript-newtype": "^3.0.0",
+    "purescript-prelude": "^4.0.1",
+    "purescript-strings": "^4.0.0",
+    "purescript-tuples": "^5.0.0",
+    "purescript-unicode": "^4.0.0",
+    "purescript-numbers": "^6.0.0",
+    "purescript-decimals": "sharkdp/purescript-decimals#8ad1954ccd6cc4cbbca76e5772f802e12c387ce8"
```
Comment (paulyoung) [inline bower.json]: We should also wait on this.

File: bower.json [hunk 3]
```diff
@@ -12,31 +12,29 @@
...
   },
   "devDependencies": {
-    "purescript-psci-support": "^3.0.0",
-    "purescript-spec": "^2.0.0",
-    "purescript-spec-discovery": "^2.0.0"
+    "purescript-psci-support": "^4.0.0",
+    "purescript-spec": "^3.0.0",
+    "purescript-spec-discovery": "owickstrom/purescript-spec-discovery#06d37dbdb2f456a95b68543cc8a0371dbebf87b5"
```
Comment (paulyoung) [inline bower.json]: We might want to wait on this.

Comment (paulyoung) [reply bower.json]: https://github.com/owickstrom/purescript-spec-discovery/pull/7

File: bower.json (hunk 1 above)
Comment (paulyoung) [reply bower.json]: https://github.com/slamdata/purescript-formatters/pull/48

File: bower.json (hunk 2 above)
Comment (paulyoung) [reply bower.json]: https://github.com/sharkdp/purescript-decimals/pull/7

Comment (paulyoung) [reply bower.json]: This was merged.

File: bower.json (hunk 1 above)
Comment (natefaubion) [reply bower.json]: This was merged, so we can change this to v4.0.0.

**Pull request merged**